bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/handler/VerisureThingHandler.java

```diff
@@ -0,0 +1,355 @@
+                    }
+                    logger.debug("Schedule vth {} and event {} with delay {}", vth, eventTranslation, delay);
+                    scheduler.schedule(new EventTrigger(vth, eventTranslation), delay, TimeUnit.MILLISECONDS);
+                    delay = delay + config.getEventTriggerDelay();
+                } else {
+                    logger.debug("Thing is null!");
+                }
+            }
+        }
+    }
+
+    protected void updateTimeStamp(@Nullable String lastUpdatedTimeStamp) {
+        updateTimeStamp(lastUpdatedTimeStamp, CHANNEL_TIMESTAMP);
+    }
+
+    protected void updateTimeStamp(@Nullable String lastUpdatedTimeStamp, ChannelUID cuid) {
+        if (lastUpdatedTimeStamp != null) {
+            try {
+                logger.trace("Parsing date {} for channel {}", lastUpdatedTimeStamp, cuid);
+                ZonedDateTime zdt = ZonedDateTime.parse(lastUpdatedTimeStamp);
+                ZonedDateTime zdtLocal = zdt.withZoneSameInstant(ZoneId.systemDefault());
+                logger.trace("Parsing datetime successful. Using date. {}", new DateTimeType(zdtLocal));
+                updateState(cuid, new DateTimeType(zdtLocal));
+            } catch (IllegalArgumentException e) {
+                logger.warn("Parsing date failed: {}.", e.getMessage(), e);
+            }
+        } else {
+            logger.debug("Timestamp is null!");
+        }
+    }
+
+    protected void updateTimeStamp(@Nullable String lastUpdatedTimeStamp, String channel) {
+        ChannelUID cuid = new ChannelUID(getThing().getUID(), channel);
+        updateTimeStamp(lastUpdatedTimeStamp, cuid);
+    }
+
+    protected @Nullable VerisureSession getSession() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            VerisureBridgeHandler vbh = (VerisureBridgeHandler) bridge.getHandler();
+            if (vbh != null) {
+                return vbh.getSession();
+            }
+        }
+        return null;
+    }
+
+    protected void scheduleImmediateRefresh(int refreshDelay) {
+        logger.debug("scheduleImmediateRefresh on thing: {}", thing);
+        Bridge bridge = getBridge();
+        if (bridge != null && bridge.getHandler() != null) {
+            VerisureBridgeHandler vbh = (VerisureBridgeHandler) bridge.getHandler();
+            if (vbh != null) {
+                vbh.scheduleImmediateRefresh(refreshDelay);
+            }
+        }
+    }
+
+    private class EventTrigger implements Runnable {
+        private @Nullable VerisureThingHandler<?> vth;
+        private @Nullable String event;
+
+        public EventTrigger(@Nullable VerisureThingHandler<?> vth, @Nullable String event) {
+            this.vth = vth;
+            this.event = event;
+        }
+
+        @Override
+        public void run() {
+            logger.debug("Trigger Event {} on {} at time {}", event, vth, DateTime.now());
```

Comment:
Please replace by `ZonedDateTime`.

```java
logger.debug("Trigger Event {} on {} at time {}", event, vth, ZonedDateTime.now());
```

Reply:
Fixed!

Reply:
You need to create a new pull request with this commit as this pull request is merged already.